### PR TITLE
Update opentelemetry-log-collection to v0.16.0

### PR DESCRIPTION
**Pull request by djaglowski**
Underlying changes [here](https://github.com/open-telemetry/opentelemetry-log-collection/releases/tag/v0.16.0)

### CODE REVIEW

File: internal/stanza/go.sum [hunk 1]
```diff
@@ -43,18 +43,29 @@ github.com/AndreasBriese/bbloom v0.0.0-20190825152654-46b345b51c96/go.mod h1:bOv
 github.com/Azure/azure-sdk-for-go v51.1.0+incompatible/go.mod h1:9XXNKU+eRnpl9moKnB4QOLf1HestfXbmab5FXxiDBjc=
 github.com/Azure/go-ansiterm v0.0.0-20170929234023-d6e3b3328b78/go.mod h1:LmzpDX56iTiv29bbRTIsUNlaFfuhWRQBWjQdVyAevI8=
 github.com/Azure/go-autorest v14.2.0+incompatible/go.mod h1:r+4oMnoxhatjLLJ6zxSWATqVooLgysK6ZNox3g/xq24=
+github.com/Azure/go-autorest/autorest v0.9.0/go.mod h1:xyHB1BMZT0cuDHU7I0+g046+BFDTQ8rEZB0s4Yfa6bI=
```
Comment (tigrannajaryan) [inline internal/stanza/go.sum:46]: Can you do `go mod tidy`? It appears there are some unrelated changes in go.sum files.

**Review by tigrannajaryan — APPROVED** [review]

File: internal/stanza/go.sum (hunk 1 above)
Comment (djaglowski) [reply internal/stanza/go.sum:46]: Unfortunately, this is what I end up with.

I believe the difference is accounted for though, because we `k8sevent_input` was previously a separate module. It had been the last operator to be managed as its own module but is no longer separate.

**Review by bogdandrutu — APPROVED** [review]
